### ISSUE TITLE
Replacing a hut always places at lvl 1

### ISSUE THREAD

**Issue by yi-fan-song** [ldtteam/minecolonies#10294]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Are you using the latest MineColonies Version?

- [X] I am running the latest **beta/release** version of MineColonies for my Minecraft version.
I am also running the latest versions of other mods that are part of my problem.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [X] Single Player
- [ ] Multi Player

### Minecraft Version

1.21

### MineColonies Version

1.1.719

### Structurize Version

1.0.752

### Current Behavior

When placing a hut that was level 2 or more, the hut only gets built as a level 1 hut.

### Expected Behavior

The hut gets built as the original level.

### Reproduction Steps

1. Level a builder hut to level 2.
2. Deconstruct it in the upgrade UI of a building.
3. After the builder has deconstructed the building, you pick up the hut block in the same interface. This gives you the hut block in your inventory.
4. Place the hut with the build tool
5. After placing the hut block in the new location, build building from the build menu. The hut gets constructed as a level 1 hut.

### Logs

https://gist.github.com/yi-fan-song/186f8db2ff94a635a6b5b501ef389900

### Anything else?

I tested this with the builder's hut in pagoda style. 

And this was tested in the direwolf 20 modpack for 1.21, manually updated minecolonies and structurize.

### Footer

---
## Viewers
* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Estyria** [comment]
Not the first report I've seen in the discord, just the first one willing to test it extensively. They were able to reproduce it as not being a one-off

**Comment by Mikec78660** [comment]
I believe I seen this as well. I thought maybe I messed up and didn't select level 5 (I was moving a level 5 guard tower). before placing the hut, didn't know if I needed to or not but was pretty sure I did. And got a level 1 building.

**Comment by Corbi84** [comment]
Can confirm, experienced the same issue with a lvl 2 university on version .274 in the ATM10 modpack.

**Comment by Baltrazx** [comment]
just had it with 2 buildings on lvl 5 aswell

**Comment by chriskeena** [comment]
Reported a very similar issue about 2 weeks ago. A dev closed the issue saying everything was normal.

**Comment by MotionlessTrain** [comment]
> Reported a very similar issue about 2 weeks ago. A dev closed the issue saying everything was normal.

I think that issue got misinterpreted, about the upgrade button being gone
Which is also why this is not marked as duplicate

**Comment by Mijithemidge** [comment]
Had it today, 12/1/2024, version 1.1.806. Trying to rotate a level 2 university (didn't account for the library attachment for mideval spruce) and it came out level 1.

**Comment by blitzingeagle** [comment]
Got the same issue. Not sure why the devs keep marking these as completed without explanation

**Comment by Talyda** [comment]
> Got the same issue. Not sure why the devs keep marking these as completed without explanation

What version of the mod are you using? I tested it 2 days ago and it is working normally in my test game

**Comment by blitzingeagle** [comment]
> > Got the same issue. Not sure why the devs keep marking these as completed without explanation
> 
> What version of the mod are you using? I tested it 2 days ago and it is working normally in my test game

I'm using 1.1.817-1.21.1, which is being used by FTB Presents Direwolf20 1.21. Any hut block that gets deconstructed, picked up, and placed with build tool does not retain the level.

**Comment by Talyda** [comment]
> > > Got the same issue. Not sure why the devs keep marking these as completed without explanation
> > 
> > 
> > What version of the mod are you using? I tested it 2 days ago and it is working normally in my test game
> 
> I'm using 1.1.817-1.21.1, which is being used by FTB Presents Direwolf20 1.21. Any hut block that gets deconstructed, picked up, and placed with build tool does not retain the level.

Yeah, if you want to test if a thing is fixed, use latest versionn =) You're quite out of date

**Comment by blitzingeagle** [comment]
> > > > Got the same issue. Not sure why the devs keep marking these as completed without explanation
> > > 
> > > 
> > > What version of the mod are you using? I tested it 2 days ago and it is working normally in my test game
> > 
> > 
> > I'm using 1.1.817-1.21.1, which is being used by FTB Presents Direwolf20 1.21. Any hut block that gets deconstructed, picked up, and placed with build tool does not retain the level.
> 
> Yeah, if you want to test if a thing is fixed, use latest versionn =) You're quite out of date

May I know around which version the fix was applied? so I can tell the server admin to update to an appropriate version

**Comment by Talyda** [comment]
Not a clue sorry! You could read through the changelogs to see - generally we recommend the latest version, though at the moment there is a problem with builders not being able to finish buildings if they have paintings, or items in item frames. 
So up to you if you want to update really!

**Comment by blitzingeagle** [comment]
Just to offer a conclusion to this here, just confirmed through tests that 1.1.846 works (did not check to see what the earliest version it has working is). For my case it's up to FTB to update the modpack with a newer version.
